src/internal/connector/sharepoint/list.go

```diff
@@ -0,0 +1,452 @@
+				errs = errors.Wrap(err, support.ConnectorStackErrorTrace(err))
+			}
+
+			itm.SetDriveItem(dItem)
+
+			fields, err := newPrefix.Fields().Get(ctx, nil)
+			if err != nil {
+				errs = errors.Wrap(err, support.ConnectorStackErrorTrace(err))
+			}
+
+			itm.SetFields(fields)
+
+			itms = append(itms, itm)
+		}
+
+		if resp.GetOdataNextLink() == nil {
+			break
+		}
+
+		builder = items.NewItemsRequestBuilder(*resp.GetOdataNextLink(), gs.Adapter())
+	}
+
+	if errs != nil {
+		return nil, errs
+	}
+
+	return itms, nil
+}
+
+// fetchColumns utility function to return columns from a site.
+// An additional call required to check for details concerning the SourceColumn.
+// For additional details:  https://learn.microsoft.com/en-us/graph/api/resources/columndefinition?view=graph-rest-1.0
+func fetchColumns(
+	ctx context.Context,
+	gs graph.Service,
+	identifier string,
+	cb *columns.ColumnsRequestBuilder,
+) ([]models.ColumnDefinitionable, error) {
+	var (
+		builder = cb
+		errs    error
+		cs      = make([]models.ColumnDefinitionable, 0)
+	)
+
+	for {
+		resp, err := builder.Get(ctx, nil)
+		if err != nil {
+			return nil, support.WrapAndAppend(support.ConnectorStackErrorTrace(err), err, errs)
+		}
+
+		for _, entry := range resp.GetValue() {
+			source, err := gs.Client().
+				SitesById(identifier).
+				ColumnsById(*entry.GetId()).
+				SourceColumn().
+				Get(ctx, nil)
+			if err != nil {
+				errs = support.WrapAndAppend(
+					"loadColumn unable to retrieve source: "+support.ConnectorStackErrorTrace(err),
+					err,
+					errs,
+				)
+
+				continue
+			}
+
+			entry.SetSourceColumn(source)
+
+			cs = append(cs, entry)
+		}
+
+		if resp.GetOdataNextLink() == nil {
+			break
+		}
+
+		builder = columns.NewColumnsRequestBuilder(*resp.GetOdataNextLink(), gs.Adapter())
+	}
+
+	return cs, nil
+}
+
+// fetchContentColumns
+// Same as fetchColumns. However, the Get() function calls are from different libraries.
+func fetchContentColumns(
+	ctx context.Context,
+	gs graph.Service,
+	identifier string,
+	cb *tc.ColumnsRequestBuilder,
+) ([]models.ColumnDefinitionable, error) {
+	var (
+		builder = cb
+		errs    error
+	)
+
+	cs := make([]models.ColumnDefinitionable, 0)
+
+	for {
+		resp, err := builder.Get(ctx, nil)
+		if err != nil {
+			return nil, support.WrapAndAppend(support.ConnectorStackErrorTrace(err), err, errs)
+		}
+
+		for _, entry := range resp.GetValue() {
+			source, err := gs.Client().
+				SitesById(identifier).
+				ColumnsById(*entry.GetId()).
+				SourceColumn().
+				Get(ctx, nil)
+			if err != nil {
+				errs = support.WrapAndAppend(
+					"load Content Column unable to retrieve source: "+support.ConnectorStackErrorTrace(err),
+					err,
+					errs,
+				)
+
+				continue
+			}
+
+			entry.SetSourceColumn(source)
+
+			cs = append(cs, entry)
+		}
+
+		if resp.GetOdataNextLink() == nil {
+			break
+		}
+
+		builder = tc.NewColumnsRequestBuilder(*resp.GetOdataNextLink(), gs.Adapter())
+	}
+
+	return cs, nil
```

Comment:
These are identical. Unfortunately, the `Get()` functions are library dependent. @ashmrtn If you know a good way to combine these functions, I would greatly appreciate it.

Reply:
```go
// define
type columnGetter func(ctx context.Context) (models.ColumnDefinitionCollectionResponseable, error)

// swap the func param libraries for this
func fetchColumns(
	ctx context.Context,
	gs graph.Service,
	identifier string,
	get columnGetter,
) ([]models.ColumnDefinitionable, error) {

// call the func with an anonymous func
fetchColumns(
  ctx, gs, id,
  func(ctx context.Context) (models.ColumnDefinitionCollectionResponseable, error) {
    return cb.Get(ctx, nil)
  })
```

Reply:
other option would probably be to define an interface that would allow getting the request and return an interface that implements `Get()` for some return type. The caller would instantiate and pass the struct implementation of the interface into the function that does the iteration. Something like

```
type RequestBuilder interface {
  BaseRequest(<common params>) RequestGetter
  PageRequest(link string) RequestGetter
}

type RequestGetter interface {
  Get(ctx, <params type>) (models.ColumnDefinitionCollectionResponseable, error)
}
```

That assumes the `<params type>` is the same for both. If it's not then either `any` and a type assertion/type cast would be needed in the struct implementation of the interface or generics would be needed in some way

Also `RequestGetter` is probably a bad name because it's slightly confusing but I don't know of a better one right now heh

Reply:
The params type are not the same for both.
```go
func (*columns.ColumnsRequestBuilder).Get(ctx context.Context, requestConfiguration *columns.ColumnsRequestBuilderGetRequestConfiguration)
```
vs.
```go
func (*tc.ColumnsRequestBuilder).Get(ctx context.Context, requestConfiguration *tc.ColumnsRequestBuilderGetRequestConfiguration) (models.ColumnDefinitionCollectionResponseable, error)
```
They are two different packages. They do the same thing. However, they are not interchangeable.

Reply:
I'm going to refactor as I can, but both of the approaches don't seem to work well with the Microsoft libraries.

Reply:
Changes in [017416a](https://github.com/alcionai/corso/pull/1633/commits/017416a93c48cb0dcd88b052304e004617b91e14)
@ryanfkeepers @ashmrtn, I will let these changes sit as I work on an attempt to upgrade the version of the sdk as the lack of updates are causing some errors.

Reply:
@dadams39 I think you're going to take on more overhead by switching contexts and affecting changes otherwise.  Since it looks like the primary churn here involves querying facets of a list (columns, etc), how about first taking my suggestion to stub those parts out for this PR?  You can get the boilerplate list retrieval into place- which is less code in flight/maintenance while you handle library updates- and handle the column retrieval implementation and refinements in a follow-up PR.